### PR TITLE
Decrease pub visibility of `scylla-cql` definitions

### PR DESCRIPTION
**Pull request by muzarski**
## Motivation

Some definitions in `scylla-cql` crate need to be public, so the `scylla` crate can make use of them. However, currently all of the definitions are implicitly re-exported outside and can be seen by the driver users. The idea is to explicitly re-export the types/functions that should be visible outside (and leave the rest of the public definitions for the internal usage of `scylla` crate).

## re-exported types
The main difficulty was to decide on what types should be re-exported publicly. The types that got re-exported can be seen in the documentation (`cargo doc --open`).

### Error types
I wasn't sure whether error types such as `scylla_cql::frame::frame_errors::FrameError` should be public. As of now, they are left public but this can be further discussed.

## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [ ] I added relevant tests for new features and bug fixes.
- [ ] All commits compile, pass static checks and pass test.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [ ] I have provided docstrings for the public items that I want to introduce.
- [ ] I have adjusted the documentation in `./docs/source/`.
- [ ] I added appropriate `Fixes:` annotations to PR description.

### CODE REVIEW

**Review by wprzytula — APPROVED** [review]
Looks like a careful step in the right direction.
This PR only hides some definitions, without modyfing any import paths - which is good from the aspect of not breaking existing code.

For comparison, please see my approach to module refactor, which additionally restructures import paths in a way that makes more sense for me:
https://github.com/wprzytula/scylla-rust-driver/blob/restructure-modules/scylla/src/cql/mod.rs

File: scylla/benches/benchmark.rs [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 use bytes::BytesMut;
-use scylla::{
-    frame::types,
-    transport::partitioner::{calculate_token_for_partition_key, Murmur3Partitioner},
+use scylla::transport::partitioner::{calculate_token_for_partition_key, Murmur3Partitioner};
+use scylla_cql::{
+    frame::{response::result::ColumnType, types},
+    types::serialize::row::SerializedValues,
 };
-use scylla_cql::{frame::response::result::ColumnType, types::serialize::row::SerializedValues};
```
Comment (Lorak-mmk) [inline scylla/benches/benchmark.rs:8]: Why is this change necessary? Apart from formatting / grouping, I see only 1 change - importing `scylla_cql::frame::types` instead of `scylla::frame::types`. but I see in `lib.rs` that `scylla::frame::types` still exists, so why change this file at all?

**Review by Lorak-mmk — CHANGES_REQUESTED** [review]

File: scylla/benches/benchmark.rs (hunk 1 above)
Comment (wprzytula) [reply scylla/benches/benchmark.rs:8]: @muzarski

Comment (muzarski) [reply scylla/benches/benchmark.rs:8]: `scylla_cql::frame::types` has a lot of utility functions such as `read/write_int`. These should not be public. OTOH, it contains the definitions of `Consistency` and `SerialConsistency` - these obviously should be public. This is the change we introduce to `scylla/src/lib.rs`.

Why do we need the change in `scylla/benches/benchmark.rs`? The answer is that we cannot use private definitions of `scylla` crate there. Benchmarks use some of the utility functions that I just mentioned above (which were just made private). This is why we change `scylla::frame::types` to `scylla_cql::frame::types` (in this crate, these utility functions are public).

**Review by Lorak-mmk — APPROVED** [review]

**Comment by muzarski** [comment]
v2: resolved conflicts with main

File: scylla/src/lib.rs [hunk 2]
```diff
@@ -102,7 +102,30 @@ pub mod macros;
...
+    pub mod types {
+        pub use scylla_cql::frame::types::{Consistency, SerialConsistency};
+    }
+
+    pub mod response {
+        pub use scylla_cql::frame::response::cql_to_rust;
+        pub(crate) use scylla_cql::frame::response::*;
+
+        pub mod result {
+            pub(crate) use scylla_cql::frame::response::result::*;
```
Comment (wprzytula) [inline scylla/src/lib.rs:121]: I'm not sure if I'm happy with the glob imports here. Being `pub(crate)` ones, they are relatively benign though.

**Review by wprzytula — APPROVED** [review]
